### ISSUE TITLE
Apache license violation

### ISSUE THREAD

**Issue by zihotki** [rsdn/CodeJam#30]
As I can see the project is released under MIT license, but some of the source files (ObjectPool.cs for example) are under Apache license and it's not mentioned anywhere and that's a violation.

**Comment by ig-sinicyn** [comment]
@zihotki 

Great point! 

Well, technically there is no problem as MIT license does allow to mix code under different licenses and we have ensured that all sources issued under Apache license have a license info applied.

We definitely should have a notice and [here's a stub for it](https://github.com/rsdn/CodeJam/blob/master/NOTICE.md). Should be improved, though.

AFAIR there is no common guideline for mixing two licenses in same project. If understand the Apache license literally, our code is fine. Quoting the Apache license
>(d) If the Work includes a "NOTICE" text file as part of its
          distribution, then any Derivative Works that You distribute must
          include a readable copy of the attribution notices contained
          within such NOTICE file
and
>APPENDIX: How to apply the Apache License to your work.
>To apply the Apache License to your work, attach the following boilerplate notice ...

The first is easy one as the Roslyn does not contain Notice document and the second is done already.

**Proofs:**
https://www.quora.com/Is-the-MIT-license-compatible-with-the-Apache-License-Version-2-APLv2
http://softwareengineering.stackexchange.com/questions/297217/using-code-under-apache-2-0-license-in-my-program-using-mit-license


Any ideas what to do further?

2team: do we need the ObjectPool part at all? 😕

**Comment by zihotki** [comment]
Also to not forget, the package on NuGet should also be updated somehow to include the notice. But I'm not sure how they suggest to solve such cases.

**Comment by ig-sinicyn** [comment]
@zihotki 
Well, here is what we've done so far:

* We removed the code issued under Apache license. It turns out the `ObjectPool` was included as a stub implementation and was not used at all.

* I've added 'Notices' section to the readme.md listing all places that refer to third-party code and checked that all such files have a license section on top of it,

If that's ok I'm going to close the issue.
